pyresample/utils/cf.py

```diff
@@ -0,0 +1,497 @@
+            cf_info[axis][k] = axis_info[axis][k]
+
+    # sanity check: we cannot have different units for x and y
+    unit = axis_info['x']['unit']
+    if axis_info['x']['unit'] != axis_info['y']['unit']:
+        raise ValueError("Cannot have different units for 'x' ({}) and 'y' ({}) axis.".format(
+            axis_info['x']['unit'], axis_info['y']['unit']))
+
+    # prepare the AreaDefinition object
+    # =================================
+
+    # create shape
+    shape = (axis_info['y']['nb'], axis_info['x']['nb'])
+
+    # get area extent from the x and y info
+    extent = _get_area_extent_from_cf_axis(axis_info['x'], axis_info['y'])
+
+    # transform the crs objecto a proj_dict (might not be needed in future versions of pyresample)
+    proj_dict = crs.to_dict()
+
+    # finally prepare the AreaDefinition object
+    area_def = geometry.AreaDefinition.from_extent('from_cf', proj_dict, shape, extent,
+                                                   units=unit, )
+
+    return area_def, cf_info
+
+
+def _load_cf_area_severalVariables(nc_handle, ):
+    """ load several AreaDefitions from a netCDF/CF file """
+
+    def _indices_unique_AreaDefs(adefs):
+        """ use a classic dict-based strategy to
+            get the indices of unique AreaDefinitions in a list """
+
+        uniqs = dict()
+        for i, adef in enumerate(adefs):
+            if not adef in uniqs:  # this uses AreaDefinition.__eq__()
+                uniqs[adef] = i
+
+        # return only the indices
+        return uniqs.values()
+
+    adefs = []
+    infos = []
+
+    # go through all the variables
+    for v in nc_handle.variables.keys():
+
+        # skip variables that are less than 2D: they cannot
+        #   possibly sustain an AreaDefinition
+        if nc_handle[v].ndim < 2:
+            continue
+
+        try:
+            # try and load an AreaDefinition from this variable
+            adef, info = _load_cf_area_oneVariable(nc_handle, v)
+            # store
+            adefs.append(adef)
+            infos.append(info)
+            # break the loop, we have all we need
+            break
+        except ValueError:
+            # this is not a problem: variable v simply doesn't define an AreaDefinition
+            continue
+
+    # go through the loaded AreaDefinitions and find the unique ones.
+    indices = _indices_unique_AreaDefs(adefs)
+    uniq_adefs = [adefs[ui] for ui in indices]
+    uniq_infos = [infos[ui] for ui in indices]
+
+    return uniq_adefs, uniq_infos
+
+
+def load_cf_area(nc_file, variable=None, y=None, x=None, with_cf_info=False):
+    """Load an AreaDefinition object from a netCDF/CF file
+
+    Parameters
+    ----------
+    nc_file : string or object
+        path to a netCDF/CF file, or opened xarray.Dataset object
+    variable : string, optional
+        name of the variable to load the AreaDefinition from
+        If variable is None the file will be searched for valid CF
+        area definitions
```

Comment:
Maybe say that if the variable itself doesn't contain the grid mapping, the variable pointed too will be read.

Reply:
good idea. done here: https://github.com/TomLav/pyresample/commit/0f5e6fd9cc2d8ed63f29fcbfc5424bf9c7efbfae